ground/src/main/java/com/google/android/ground/ui/home/HomeScreenFragment.kt

```diff
@@ -173,21 +173,33 @@ class HomeScreenFragment :
     // compose.
     binding.composeView.apply {
       setContent {
-        val showUserDetailsDialog = remember { mutableStateOf(true) }
+        val showUserDetailsDialog = remember { mutableStateOf(false) }
         val showSignOutDialog = remember { mutableStateOf(false) }
 
-        // reset the state for recomposition
-        showUserDetailsDialog.value = true
-        showSignOutDialog.value = false
+        fun showUserDetailsDialog() {
+          showUserDetailsDialog.value = true
+          showSignOutDialog.value = false
+        }
+
+        fun showSignOutDialog() {
+          showUserDetailsDialog.value = false
+          showSignOutDialog.value = true
+        }
+
+        fun hideAllDialogs() {
+          showUserDetailsDialog.value = false
+          showSignOutDialog.value = false
+        }
```

Comment:
can we manage this using state rather then nested function?
WDYT?

Reply:
I haven't changed the business logic. Just created new helper methods for improved readability. What exactly are you proposing? Can you please elaborate?

Reply:
Rather than having these three nested functions which maintains the `value - true/false`, can we use, some state for managing it?

```kt
showUserDetails -> { }
showSignOut -> { }
hideAll -> { }
```

Reply:
We already have 2 states: `showSignOutDialog` and `showUserDetailsDialog` which are bound to the dialogs. Are you proposing to replace those as well?

Reply:
no, I mean removing these functions

```kt
fun showUserDetailsDialog() {}
fun showSignOutDialog() {}
fun hideAllDialogs() {}
```

and updating the states based on some single source state or any stateflow which can emit some data telling what to make false, what to make true
```
 showUserDetailsDialog.value = true
          showSignOutDialog.value = false

   showUserDetailsDialog.value = false
          showSignOutDialog.value = true

showUserDetailsDialog.value = false
          showSignOutDialog.value = false
```